lnbits/wallets/cliche.py

```diff
@@ -157,19 +184,30 @@ async def get_payment_status(self, checking_id: str) -> PaymentStatus:
             payment.get("preimage"),
         )
 
+    async def on_message(self, message: str):
+        try:
+            data = json.loads(message)
+            msg_id = data.get("id")
+            if msg_id is not None and msg_id in self.futures:
+                future = self.futures.pop(msg_id)
+                future.set_result(data)
+            elif data.get("result", {}).get("status"):
+                yield data["result"]["payment_hash"]
+        except Exception as exc:
+            logger.exception(f"Error processing message: {exc}")
+    
+    def on_error(self, error):
+        logger.warning(f"Error from websocket: {error}")
+    
+    def on_close(self):
+        logger.error("Websocket closed")
+        self.ws.close()
```

Comment:
when is this triggered?

Reply:
here (line 33):
`async def connect(self):
        self.ws = WebSocketApp(
            self.endpoint,
            on_message=self.on_message,
            on_error=self.on_error,
            on_close=self.on_close,
        )`